File: python-lang/8_kyu/do_you_speak_english_test.py
import pytest

from do_you_speak_english import sp_eng


@pytest.mark.parametrize(
    "sentence,expected",
    [
        ("english", True),
        ("egnlish", False),
        ("engliish", False),
        ("1234egn lis;h", False),
        ("1234english ;k", True),
        ("English", True),
        ("eNgliSh", True),
        ("1234#$%%eNglish ;k9", True),
        ("EGNlihs", False),
        ("1234englihs**", False),
        ("", False),
        (None, False)
    ]
)
def test_example_cases(sentence: str, expected: bool):
    assert sp_eng(sentence) is expected

File: typescript-lang/8-kyu/do-you-speak-english.test.ts
import { spEng } from "./do-you-speak-english.ts";
import { assertEquals } from "@std/assert";

Deno.test.each([
  {
    sentence: "english",
    expected: true,
  },
  { sentence: "egnlish", expected: false },
  { sentence: "engliish", expected: false },
  { sentence: "1234egn lis;h", expected: false },
  { sentence: "1234english ;k", expected: true },
  { sentence: "English", expected: true },
  { sentence: "eNgliSh", expected: true },
  { sentence: "1234#$%%eNglish ;k9", expected: true },
  { sentence: "EGNlihs", expected: false },
  { sentence: "1234englihs**", expected: false },
  { sentence: "", expected: false },
  { sentence: undefined, expected: false },
  { sentence: null, expected: false },
])(
  "should return false or true when sentence compare to 'english'",
  ({ sentence, expected }): void => {
    assertEquals(spEng(sentence), expected);
  },
);
